app/src/main/java/com/example/lemonade/MainActivity.kt

```diff
@@ -19,44 +19,171 @@ package com.example.lemonade
+                        }
+                    )
+                }
+                2 -> {
+                    LemonTextAndImage(
+                        textLabelResourceId = R.string.lemon_squeeze,
+                        drawableResourceId = R.drawable.lemon_squeeze,
+                        contentDescriptionResourceId = R.string.lemon_content_description,
+                        onImageClick = {
+                            squeezeCount--
+                            if (squeezeCount == 0) {
+                                currentStep = 3
+                            }
+                        }
+                    )
+                }
+
+                3 -> {
+                    LemonTextAndImage(
+                        textLabelResourceId = R.string.lemon_drink,
+                        drawableResourceId = R.drawable.lemon_drink,
+                        contentDescriptionResourceId = R.string.lemonade_content_description,
+                        onImageClick = {
+                            currentStep = 4
+                        }
+                    )
+                }
+                4 -> {
+                    LemonTextAndImage(
+                        textLabelResourceId = R.string.lemon_empty_glass,
+                        drawableResourceId = R.drawable.lemon_restart,
+                        contentDescriptionResourceId = R.string.empty_glass_content_description,
+                        onImageClick = {
+                            currentStep = 1
+                        }
+                    )
                 }
             }
         }
     }
 }
 
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-    Text(
-        text = "Hello $name!",
-        modifier = modifier
-    )
+fun LemonTextAndImage(
+    textLabelResourceId: Int,
+    drawableResourceId: Int,
+    contentDescriptionResourceId: Int,
+    onImageClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+        modifier = modifier.fillMaxSize()
+    ) {
+
+        Button(
+            onClick = onImageClick,
+            shape = RoundedCornerShape(dimensionResource(R.dimen.button_corner_radius)),
+            colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.tertiaryContainer)
+        ) {
+            Image(
+                painter = painterResource(drawableResourceId),
+                contentDescription = stringResource(contentDescriptionResourceId),
+                modifier = Modifier
+                    .wrapContentSize()
+                    .padding(dimensionResource(R.dimen.button_interior_padding))
+            )
+        }
+        Spacer(modifier = Modifier.height(dimensionResource(R.dimen.padding_vertical)))
+        Text(
+            text = stringResource(textLabelResourceId),
+            fontSize = 18.sp
```

Comment:
assign to something in Type.kt and assign it here instead of setting the fontSize